muskrat/s3consumer.py

```diff
@@ -125,12 +142,8 @@ def consume(self):
         #Update: actually... this doesn't seem to be a problem...
         msg_iterator = self._get_msg_iterator()
 
-        for msg in msg_iterator:
-            #Sub 'directories' are prefix objects, so ignore them
-            if isinstance( msg, boto.s3.key.Key ):
-                self.callback( msg.get_contents_as_string() )
-                self._cursor.update( msg.name )
-
+        for obj in self._cursor.each(msg_iterator):
+            self.callback(obj.get()['Body'].read())
```

Comment:
What is the behavior of this if the obj/key is not message.  If this key points to something that is considered a namespace (something like a 'sub directory'.  Not sure what to call it in S3) then what happens?